### ISSUE TITLE
Problem with Date format in add_chunk : error "the belows types are incompatible between the new chunk and the disk.frame"

### ISSUE THREAD

**Issue by palamito** [xiaodaigh/disk.frame#292]
Hi,

Thank you for your excellent package. I am currently encountering a problem with the function add_chunk when adding chunks to a disk.frame with some variables in "Date" class.

For instance, let's say I have 3 dataframes with the same columns. One of them has Date format (here below, "period"). While writing the first chunk is doing fine, the second one will throw the following error :

```
the belows types are incompatible between the new chunk and the disk.frame; this chunk can not be added

   colnames coltypes.x new_chunk coltypes.y N existing_df incompatible_types
1:   period     double      TRUE       Date 1        TRUE               TRUE

Error in add_chunk(main.disk.frame, get(list[i]), chunk_id = id[i], compress = 100) :
```

It seems like that the variable "period" has two different formats in data frame x and date frame y. However, I double checked the original format of my data frames and I confirm that the original format is Date.

Could you explain why ? Has this something to do with the binary compression you are using in disk.frame ?

Thanks

**Comment by xiaodaigh** [comment]
The error message tells me that one of them is double and one is Date. Maybe the storage messed up the format. 

Is there a way I can get the data or you can generate some synthetic data with code that can replicate this issue?

**Comment by xiaodaigh** [comment]
I can replicate the error with

```r

library(disk.frame)
setup_disk.frame()

a = data.frame(a = as.Date("2020-07-01"), b = runif(1e6))

a.df = as.disk.frame(a)

head(a.df)


a.df = add_chunk(a.df, a)
```

Looking into it now.

**Comment by xiaodaigh** [comment]
@palamito Thank you very much for you bug report.

I have now fixed the bug and added a test to prevent regression. New package on way to CRAN.

**Comment by palamito** [comment]
Thank you @xiaodaigh . It works brilliantly !

**Comment by xiaodaigh** [comment]
This bug has been fixed in the latest CRAN release on v0.3.7. Takes about 1 week for it to be available for all platforms with binaries.
